### PR TITLE
thumb : rating class + allow to fade the image drawing

### LINKED ISSUE #9518: Adjust thumbnail borders based on rating/rejection
**Describe the solution you'd like**

I like the clutterless look of the thumbnails in lighttable and darkroom mode. But at the same time I want to edit and cull the photos while keeping my eye on the rejected ones. Ideally I would want to see darkened or otherwise styled borders for thumbnails of rejected images.

**Alternatives**

I can always activate the "permament overlays" in lighttable module. It won't have an effect in darkroom film roll, though.

**Additional context**

The custom CSS rules was my initial place to venture but the CSS3 selectors aren't flexible enough to allow this modification.

I don't know GTK but if these UI elements + CSS work like the web browser equivalents, perhaps it would be trivial to add some suitable classes to thumbnail elements and let the rest up to the stylesheets.

### PR DESCRIPTION
**Pull request by AlicVB**
this fix #9518 this close #9532 

this add a class to the main thumb widget to expose the rating value : `.dt_thumbnail_rating_[0-6]` where 6 means that the image is rejected (the numbers comes from dt internal code already in place)
this way one can tweak all css of the different thumb widgets depending on the rating value.

this also add a way to draw the thumb image with a transparency value, defined in the `color` property of the `thumb_image` widget. Note that only the transparency part of the property is handled (this is kind of a hack, but I've not found a better way to define that in css)

@Nilvus : feel free to add/propose some new css or change the transparency value, I've put 0.5 more as an example than anything else...

### CODE REVIEW

File: data/themes/darktable.css
Comment (TurboGit) [inline data/themes/darktable.css:2259]: I would even use .25 for the alpha channel to make it easier to see rejected images.

**Review by TurboGit — COMMENTED** [review]
Working for me and looks good a minor comment about alpha channel.

File: data/themes/darktable.css
Comment (AlicVB) [reply data/themes/darktable.css:2259]: no problem... as said, I've put 0.5 more as an example than anything...
maybe we can wait for @Nilvus review, as I'm quite sure he will have some more css finetuning ?

Comment (TurboGit) [reply data/themes/darktable.css:2259]: Sure, it was a pure comment and I had added @Nilvus as reviewer for him to have the last word on this :)

Comment (Nilvus) [reply data/themes/darktable.css:2259]: Just a quick test and maybe something 0.25 and 0.5 would be better. Seems 0.35 is something good. Anyway, I don't have a lot of time this week to test further. Will try to check that on Wednesday, or if not possible, only next week.

Comment (Nilvus) [reply data/themes/darktable.css:2259]: I would be for 0.35, seems a good compromise. I also think this would be good too to fade other parts of rejected images (ext, group/altered icon and also star when hovering). But all this could be done in a separate PR by me. I so suggested to simply update to 0.35 here and merge if all good for you.

**Review by Nilvus — APPROVED** [review]
Except my last comment to update fade from 0.5 to 0.35, it's all good for me

**Comment by AlicVB** [comment]
go for 0.35 then :) thanks for your tests !
PR updated

**Comment by TurboGit** [comment]
@AlicVB : I think we should also dim the thumb background (around the image). The commit I've done in my previous PR should be ok. So I'll investigate this part if you don't mind.

See 5073b6fe9a618f2b7e402a930a1f1c9995c03da8.

**Comment by AlicVB** [comment]
@TurboGit : with this PR, it's really easy and don't need the commit you point anymore : you just have to set the wanted css background-color for `.dt_thumbnail_rating_6 #thumb_back`
(I hope I've understood correctly what you want to achieve...)

**Comment by Nilvus** [comment]
I agree with @AlicVB. With that PR, all thumbnails parts could be set and so fade the thumb background too. I will work on that after this is merged and last week.

**Comment by TurboGit** [comment]
@Nilvus : Indeed, looking at the code I see that nothing more is needed except some CSS work. I let you do that. TIA.

**Review by TurboGit — APPROVED** [review]
All good, thanks!

**Pull request merged**
